edb/server/pgcon/pgcon.pyx

```diff
@@ -2057,6 +2080,44 @@ cdef class PGConnection:
                 ):
                     self.buffer.consume_message()
 
+                elif (
+                    mtype == b'T'  # RowDescription
+                    and action.action == PGAction.EXECUTE
+                    and isinstance(
+                        action.query_unit.command_complete_tag,
+                        dbstate.TagUnpackRow,
+                    )
+                ):
+                    # when unpacking a DataRow, we need to know if it is in
+                    # in binary or text mode.
+                    # Here we also assume that there will be a single column
+                    # in the DataRow.
+                    is_binary_format = int.from_bytes(data[-2:0], "big") == 1
```

Comment:
`data` may be an undefined/unassigned variable? And I'm confused by the `[-2:0]` slicing - that'll make `from_bytes()` always return `0` I suppose

Reply:
Yes, this chunk was a bad copy-paste, thanks.